dask/tests/test_config.py

```diff
@@ -161,22 +160,23 @@ def test_env():
     assert res == expected
 
 
-def test_collect():
+def test_collect(tmp_path_factory):
     a = {"x": 1, "y": {"a": 1}}
     b = {"x": 2, "z": 3, "y": {"b": 2}}
     env = {"DASK_W": 4}
 
     expected = {"w": 4, "x": 2, "y": {"a": 1, "b": 2}, "z": 3}
 
-    with tmpfile(extension="yaml") as fn1:
-        with tmpfile(extension="yaml") as fn2:
-            with open(fn1, "w") as f:
-                yaml.dump(a, f)
-            with open(fn2, "w") as f:
-                yaml.dump(b, f)
+    fn1 = os.path.join(tmp_path_factory.mktemp("dn"), "fn.yaml")
+    fn2 = os.path.join(tmp_path_factory.mktemp("dn"), "fn.yaml")
```

Comment:
Do we need to create a separate subdirectory here? I think we can accomplish the same thing with less code using `tmp_path`

```diff
diff --git a/dask/tests/test_config.py b/dask/tests/test_config.py
index c18cd242..aef3d9b7 100644
--- a/dask/tests/test_config.py
+++ b/dask/tests/test_config.py
@@ -160,15 +160,15 @@ def test_env():
     assert res == expected


-def test_collect(tmp_path_factory):
+def test_collect(tmp_path):
     a = {"x": 1, "y": {"a": 1}}
     b = {"x": 2, "z": 3, "y": {"b": 2}}
     env = {"DASK_W": 4}

     expected = {"w": 4, "x": 2, "y": {"a": 1, "b": 2}, "z": 3}

-    fn1 = os.path.join(tmp_path_factory.mktemp("dn"), "fn.yaml")
-    fn2 = os.path.join(tmp_path_factory.mktemp("dn"), "fn.yaml")
+    fn1 = os.path.join(tmp_path, "fn1.yaml")
+    fn2 = os.path.join(tmp_path, "fn2.yaml")

     with open(fn1, "w") as f:
         yaml.dump(a, f)
```

Reply:
Actually we don't need it, update the PR